ddev/ddev
# Switch to NodeJS LTS (18) by default and support NodeJS 20, fixes #4674

## The Issue

* #4674 

NodeJS 18 is now officially LTS with the NodeJS 20 release today:
https://github.com/nodejs/Release
https://nodejs.org/en/download

## Blockers

The 20 version is not available yet from nodesource distributions, but I have submitted a PR for this at https://github.com/nodesource/distributions/pull/1544

## How This PR Solves The Issue

* Updates the Docker image to use `NODE_LTS` of `18`
* ~~Adds support for the `nodejs_version` configuration to be set to `lts` (and is the default). Potentially it seems like we could maybe consider removing this configuration in favor of `ddev nvm`~~

## Manual Testing Instructions

* Restart your DDEV site with this change and running `ddev node --version` should display a version like 18.N.N
* Does this succeed?
  ```
  ddev config --nodejs-version 20
  ```
* Does this fail?
  ```
  ddev config --nodejs-version 19
  ```

## Automated Testing Overview

Should not need any changes to automated tests.

## Related Issue Link(s)

Fixes https://github.com/ddev/ddev/issues/4674

## Release/Deployment Notes





<a href="https://gitpod.io/#https://github.com/ddev/ddev/pull/4833"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
